lib/metasploit/framework/login_scanner/http.rb

```diff
@@ -186,6 +191,13 @@ class HTTP
 
         # (see Base#check_setup)
         def check_setup
+          if http_success_codes.nil?
+            @http_success_codes = DEFAULT_HTTP_SUCCESS_CODES
```

Comment:
I think this can be done as part of the set_sane_defaults logic that scanners can implement